Provide more relevant stacktrace in StepVerifier

The stacktrace printed by default by [`DefaultStepVerifierBuilder#fail()`](
https://github.com/reactor/reactor-addons/blob/master/reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java#L1643) seems not very relevant nor useful.

In my use case my test is (Kotlin code):
```
kotlin
Hooks.onOperator<Any> { h -> h.operatorStacktrace() }
StepVerifier.create(response.flatMap{ r -> r.bodyToFlux(User::class)})
                .consumeNextWith {
                    assert(it == User(1L, "Robert")) }
                .consumeNextWith { assert(it == User(2L, "Raide"))  }
                .consumeNextWith { assert(it == User(3L, "Ford")) }
                .expectComplete()
                .verify()
```

I get that output:
```
java.lang.AssertionError: expectation "consumeNextWith" failed (expected: onNext(); actual: onError(java.lang.ArrayIndexOutOfBoundsException: -1))

	at reactor.test.DefaultStepVerifierBuilder.failPrefix(DefaultStepVerifierBuilder.java:1653)
	at reactor.test.DefaultStepVerifierBuilder.fail(DefaultStepVerifierBuilder.java:1649)
	at reactor.test.DefaultStepVerifierBuilder.lambda$consumeNextWith$1(DefaultStepVerifierBuilder.java:139)
	at reactor.test.DefaultStepVerifierBuilder$SignalEvent.test(DefaultStepVerifierBuilder.java:1409)
	at reactor.test.DefaultStepVerifierBuilder$DefaultVerifySubscriber.onSignal(DefaultStepVerifierBuilder.java:1050)
	at reactor.test.DefaultStepVerifierBuilder$DefaultVerifySubscriber.onExpectation(DefaultStepVerifierBuilder.java:996)
	at reactor.test.DefaultStepVerifierBuilder$DefaultVerifySubscriber.onError(DefaultStepVerifierBuilder.java:740)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onError(FluxOnAssembly.java:322)
	at reactor.core.publisher.MonoFlatMap$FlattenSubscriber$InnerSubscriber.onError(MonoFlatMap.java:196)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onError(FluxOnAssembly.java:322)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onError(FluxMapFuseable.java:140)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onError(FluxOnAssembly.java:322)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.checkTerminated(FluxFlatMap.java:794)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drainLoop(FluxFlatMap.java:558)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.drain(FluxFlatMap.java:538)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onError(FluxFlatMap.java:517)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:356)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:316)
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:173)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:316)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:316)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at reactor.ipc.netty.channel.FluxReceive.drainReceiver(FluxReceive.java:185)
	at reactor.ipc.netty.channel.FluxReceive.onInboundNext(FluxReceive.java:291)
	at reactor.ipc.netty.channel.ChannelOperations.onInboundNext(ChannelOperations.java:284)
	at reactor.ipc.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:493)
	at reactor.ipc.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:112)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:280)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:396)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:248)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:129)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at java.lang.Thread.run(Thread.java:745)
```

And to have the relevant stacktrace that show me the origin of the error, I have to call `signal.getThrowable().printStackTrace()` from `consumeNextWith()`:

```
java.lang.ArrayIndexOutOfBoundsException: -1
	at io.netty.buffer.HeapByteBufUtil.getByte(HeapByteBufUtil.java:24)
	at io.netty.buffer.UnpooledHeapByteBuf._getByte(UnpooledHeapByteBuf.java:323)
	at io.netty.buffer.UnpooledHeapByteBuf.getByte(UnpooledHeapByteBuf.java:318)
	at org.springframework.http.codec.json.JsonObjectDecoder$1.apply(JsonObjectDecoder.java:140)
	at org.springframework.http.codec.json.JsonObjectDecoder$1.apply(JsonObjectDecoder.java:103)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:353)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:316)
	at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:173)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:316)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:316)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at reactor.ipc.netty.channel.FluxReceive.drainReceiver(FluxReceive.java:185)
	at reactor.ipc.netty.channel.FluxReceive.onInboundNext(FluxReceive.java:291)
	at reactor.ipc.netty.channel.ChannelOperations.onInboundNext(ChannelOperations.java:284)
	at reactor.ipc.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:493)
	at reactor.ipc.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:112)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:280)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:396)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:248)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:129)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
	at java.lang.Thread.run(Thread.java:745)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Assembly trace from producer [reactor.core.publisher.FluxFlatMap] :
	reactor.core.publisher.Flux.flatMap(Flux.java:3401)
	reactor.core.publisher.Flux.flatMap(Flux.java:3378)
	reactor.core.publisher.Flux.flatMap(Flux.java:3335)
	org.springframework.http.codec.json.JsonObjectDecoder.decode(JsonObjectDecoder.java:103)
	org.springframework.http.codec.json.Jackson2JsonDecoder.decodeInternal(Jackson2JsonDecoder.java:114)
	org.springframework.http.codec.json.Jackson2JsonDecoder.decode(Jackson2JsonDecoder.java:82)
	org.springframework.http.codec.DecoderHttpMessageReader.read(DecoderHttpMessageReader.java:75)
	org.springframework.http.codec.BodyExtractors.lambda$null$2(BodyExtractors.java:91)
	java.util.Optional.map(Optional.java:215)
	org.springframework.http.codec.BodyExtractors.readWithMessageReaders(BodyExtractors.java:120)
	org.springframework.http.codec.BodyExtractors.lambda$toFlux$3(BodyExtractors.java:89)
	org.springframework.web.client.reactive.DefaultClientResponse.body(DefaultClientResponse.java:72)
	org.springframework.web.client.reactive.DefaultClientResponse.bodyToPublisher(DefaultClientResponse.java:102)
	org.springframework.web.client.reactive.DefaultClientResponse.bodyToFlux(DefaultClientResponse.java:87)
	mixit.support.ExtensionsKt.bodyToFlux(Extensions.kt:53)
	mixit.integration.UserIntegrationTests$findAll$1.apply(UserIntegrationTests.kt:31)
	mixit.integration.UserIntegrationTests$findAll$1.apply(UserIntegrationTests.kt:16)
	reactor.core.publisher.MonoFlatMap$FlattenSubscriber.onNext(MonoFlatMap.java:121)
	reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:124)
	reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:173)
	reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:124)
	reactor.core.publisher.FluxResume$ResumeSubscriber.onNext(FluxResume.java:71)
	reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:124)
	reactor.core.publisher.FluxRetryPredicate$RetryPredicateSubscriber.onNext(FluxRetryPredicate.java:78)
	reactor.core.publisher.MonoCreate$DefaultMonoSink.success(MonoCreate.java:95)
	reactor.ipc.netty.channel.PooledClientContextHandler.fireContextActive(PooledClientContextHandler.java:73)
	reactor.ipc.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:474)
	reactor.ipc.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:112)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:435)
	io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:293)
	io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:267)
	io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:250)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:86)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:351)
	io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:373)
	io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:359)
	io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
	io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:129)
	io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:651)
	io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:574)
	io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:488)
	io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:450)
	io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:873)
Composition chain until failing Operator :
	|_	Flux.flatMap(JsonObjectDecoder.java:103)
	|_	Flux.map(Jackson2JsonDecoder.java:115)
```
 It should be nice to have that information by default in the console after a failed test.

--- a/reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java
+++ b/reactor-test/src/main/java/reactor/test/DefaultStepVerifierBuilder.java
@@ -997,7 +997,6 @@ final class DefaultStepVerifierBuilder<T>
 						return;
 					}
 				}
-//				System.out.println("passe " + actualSignal + ", produced: " + produced + ", unasserted: " + unasserted);
 
 				event = this.script.peek();
 				if (event == null || !(event instanceof EagerEvent)) {
@@ -1050,6 +1049,10 @@ final class DefaultStepVerifierBuilder<T>
 			Optional<AssertionError> error = signalEvent.test(actualSignal);
 			if (error.isPresent()) {
 				Exceptions.addThrowable(ERRORS, this, error.get());
+				// #55 ensure the onError is added as a suppressed to the AssertionError
+				if(actualSignal.isOnError()) {
+					error.get().addSuppressed(actualSignal.getThrowable());
+				}
 				maybeCancel(actualSignal);
 				this.completeLatch.countDown();
 				return true;
@@ -1086,6 +1089,10 @@ final class DefaultStepVerifierBuilder<T>
 			}
 			else {
 				Exceptions.addThrowable(ERRORS, this, error.get());
+				if(actualSignal.isOnError()) {
+					// #55 ensure the onError is added as a suppressed to the AssertionError
+					error.get().addSuppressed(actualSignal.getThrowable());
+				}
 				maybeCancel(actualSignal);
 				this.completeLatch.countDown();
 				return true;
@@ -1124,6 +1131,10 @@ final class DefaultStepVerifierBuilder<T>
 
 					if (error.isPresent()) {
 						Exceptions.addThrowable(ERRORS, this, error.get());
+						if(actualSignal.isOnError()) {
+							// #55 ensure the onError is added as a suppressed to the AssertionError
+							error.get().addSuppressed(actualSignal.getThrowable());
+						}
 						maybeCancel(actualSignal);
 						this.completeLatch.countDown();
 					}
@@ -1292,10 +1303,8 @@ final class DefaultStepVerifierBuilder<T>
 				return;
 			}
 
-			if (errors.getSuppressed().length == 0){
-				if(errors instanceof AssertionError){
-					throw (AssertionError)errors;
-				}
+			if(errors instanceof AssertionError){
+				throw (AssertionError)errors;
 			}
 
 			List<Throwable> flat = new ArrayList<>();

--- a/reactor-test/src/test/java/reactor/test/StepVerifierTests.java
+++ b/reactor-test/src/test/java/reactor/test/StepVerifierTests.java
@@ -19,6 +19,7 @@ import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.LongAdder;
 
@@ -1393,5 +1394,92 @@ public class StepVerifierTests {
 	                .verifyComplete())
 	            .withMessage("expectation \"expectComplete\" failed (expected: onComplete(); actual: onNext(4))");
 	}
+
+	@Test
+	public void expectNextErrorIsSuppressed() {
+		assertThatExceptionOfType(AssertionError.class)
+				.isThrownBy(() -> StepVerifier.create(Flux.just("foo")
+				                                          .flatMap(r -> { throw new ArrayIndexOutOfBoundsException();}))
+	                .expectNext("foo")
+	                .verifyError())
+				.satisfies(error -> {
+					assertThat(error)
+							.hasMessageStartingWith("expectation \"expectNext(foo)\" failed")
+							.hasMessageContaining("actual: onError(java.lang.ArrayIndexOutOfBoundsException)");
+					assertThat(error.getSuppressed())
+							.hasSize(1)
+							.allMatch(spr -> spr instanceof ArrayIndexOutOfBoundsException);
+				});
+	}
+
+	@Test
+	public void consumeNextErrorIsSuppressed() {
+		assertThatExceptionOfType(AssertionError.class)
+				.isThrownBy(() -> StepVerifier.create(Flux.just("foo")
+				.flatMap(r -> { throw new ArrayIndexOutOfBoundsException();}))
+	                .consumeNextWith(v -> assertThat(v).isNotNull())
+	                .verifyError())
+				.satisfies(error -> {
+					assertThat(error)
+							.hasMessageStartingWith("expectation \"consumeNextWith\" failed")
+							.hasMessageContaining("actual: onError(java.lang.ArrayIndexOutOfBoundsException)");
+					assertThat(error.getSuppressed())
+							.hasSize(1)
+							.allMatch(spr -> spr instanceof ArrayIndexOutOfBoundsException);
+				});
+	}
+
+	@Test
+	public void expectNextCountErrorIsSuppressed() {
+		assertThatExceptionOfType(AssertionError.class)
+				.isThrownBy(() -> StepVerifier.create(Flux.just("foo")
+				.flatMap(r -> { throw new ArrayIndexOutOfBoundsException();}))
+	                .expectNextCount(1)
+	                .verifyError())
+				.satisfies(error -> {
+					assertThat(error)
+							.hasMessageStartingWith("expectation \"expectNextCount\" failed")
+							.hasMessageContaining("signal: onError(java.lang.ArrayIndexOutOfBoundsException)");
+					assertThat(error.getSuppressed())
+							.hasSize(1)
+							.allMatch(spr -> spr instanceof ArrayIndexOutOfBoundsException);
+				});
+	}
+
+	@Test
+	public void expectNextSequenceErrorIsSuppressed() {
+		assertThatExceptionOfType(AssertionError.class)
+				.isThrownBy(() -> StepVerifier.create(Flux.just("foo")
+				.flatMap(r -> { throw new ArrayIndexOutOfBoundsException();}))
+	                .expectNextSequence(Arrays.asList("foo"))
+	                .verifyError())
+				.satisfies(error -> {
+					assertThat(error)
+							.hasMessageStartingWith("expectation \"expectNextSequence\" failed")
+							.hasMessageContaining("actual signal: onError(java.lang.ArrayIndexOutOfBoundsException)");
+					assertThat(error.getSuppressed())
+							.hasSize(1)
+							.allMatch(spr -> spr instanceof ArrayIndexOutOfBoundsException);
+				});
+	}
+
+	@Test
+	public void consumeWhileErrorIsSuppressed() {
+		assertThatExceptionOfType(AssertionError.class)
+				.isThrownBy(() -> StepVerifier.create(Flux.just("foo", "bar", "foobar")
+		                        .map(r -> { if (r.length() > 3) throw new ArrayIndexOutOfBoundsException(); return r;}))
+	                .thenConsumeWhile(s -> s.length() <= 3) //doesn't fail by itself...
+	                .verifyComplete()) //...so this will fail
+				.satisfies(error -> {
+					assertThat(error)
+							.hasMessageStartingWith("expectation \"expectComplete\" failed")
+							.hasMessageContaining("actual: onError(java.lang.ArrayIndexOutOfBoundsException)");
+					assertThat(error.getSuppressed())
+							.hasSize(1)
+							.allMatch(spr -> spr instanceof ArrayIndexOutOfBoundsException);
+				});
+	}
+
+
 
 }

